File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.java
class Solution {
    public int solution(int n, int w, int num) {
        int div = (int) (num - 1) / w;
        int mod = (num - 1) % w;
        int x = 0;
        if (div % 2 == 0) {
            x = w - mod;
        } else {
            x = mod + 1;
        }
        
        int temp = x * 2 - 1;
        int[] step = {temp, w * 2 - temp};
        
        int answer = 0;
        while (num < n) {
            if (div % 2 == 0) {
                num += step[0];
            } else {
                num += step[1];
            }
            div++; answer++;
        }
        
        if (num == n) answer++;
        return answer;
    }
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n, w, num):
    div, mod = divmod(num - 1, w)
    x = 0
    if div % 2 == 0:
        x = w - mod
    else:
        x = mod + 1
    
    step = [x * 2 - 1, w * 2 - (x * 2 - 1)]
    
    answer = 0
    while num < n:
        if div % 2 == 0:
            num += step[0]
        else:
            num += step[1]
        div += 1
        answer += 1

    if num == n:
        answer += 1
    return answer